navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -597,6 +598,7 @@ public OffRoute getOffRouteEngine() {
 
   @Override
   public void onProgressChange(Location location, RouteProgress routeProgress) {
+    Timber.v("Arrived event occurred");
```

Comment:
Won't all of the logic in this `onProgressChange` callback get called multiple times per navigation session?

Reply:
Nope, it only gets invoked when close to the destination: https://github.com/mapbox/mapbox-navigation-android/blob/master/navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcher.java#L116-L118

Reply:
Ahhh, I was confused by the naming. Maybe we should change the internal `onProgressChange` method to `onArrival`.